modules/gapi/src/streaming/onevpl/engine/preproc/preproc_engine.cpp

```diff
@@ -0,0 +1,491 @@
+
+                    GAPI_LOG_DEBUG(nullptr, "pending VPP operations count: " <<
+                                            my_sess.vpp_out_queue.size() <<
+                                            ", sync id:  " <<
+                                            pending_op.first <<
+                                            ", surface:  " <<
+                                            pending_op.second <<
+                                            ", status: " <<
+                                            mfxstatus_to_string(my_sess.last_status));
+
+                    // put frames in ready queue on success
+                    if (MFX_ERR_NONE == sess.last_status) {
+                        on_frame_ready(my_sess, pending_op.second);
+                    }
+                }
+            } while (MFX_ERR_NONE == sess.last_status && !my_sess.vpp_out_queue.empty());
+            return ExecutionStatus::Continue;
+        },
+        // 2) Falls back on generic status procesing
+        [this] (EngineSession& sess) -> ExecutionStatus
+        {
+            return this->process_error(sess.last_status, static_cast<session_type&>(sess));
+        }
+    );
+}
+
+cv::util::optional<pp_params> VPPPreprocEngine::is_applicable(const cv::MediaFrame& in_frame) {
+    // TODO consider something smarter than RTI
+    cv::util::optional<pp_params> ret;
+    BaseFrameAdapter *vpl_adapter = in_frame.get<BaseFrameAdapter>();
+    GAPI_LOG_DEBUG(nullptr, "validate VPP preprocessing is applicable for frame");
+    if (vpl_adapter) {
+        ret = cv::util::make_optional<pp_params>(
+                        pp_params::create<vpp_pp_params>(vpl_adapter->get_session_handle(),
+                                                         vpl_adapter->get_surface()->get_info()));
+        GAPI_LOG_DEBUG(nullptr, "VPP preprocessing applicable, session [" <<
+                                vpl_adapter->get_session_handle() << "]");
+    }
+    return ret;
+}
+
+pp_session VPPPreprocEngine::initialize_preproc(const pp_params& initial_frame_param,
+                                                const GFrameDesc& required_frame_descr) {
+    const vpp_pp_params &params = initial_frame_param.get<vpp_pp_params>();
+
+    // adjust preprocessing settings
+    mfxVideoParam mfxVPPParams{0};
+    // NB: IN params for VPP session must be equal to decoded surface params
+    mfxVPPParams.vpp.In = params.info;
+
+    // NB: OUT params must refer to IN params of a network
+    GAPI_LOG_DEBUG(nullptr, "network input size: " << required_frame_descr.size.width <<
+                            "x" << required_frame_descr.size.height);
+    mfxVPPParams.vpp.Out = mfxVPPParams.vpp.In;
+    switch (required_frame_descr.fmt) {
+        case MediaFormat::NV12:
+            mfxVPPParams.vpp.Out.FourCC = MFX_FOURCC_NV12;
+            break;
+        default:
+            GAPI_LOG_WARNING(nullptr, "Unsupported MediaFormat in preprocessing: " <<
```

Comment:
it should be CRITICAL not WARNING, isn't it?

I believe @TolyaTalamanov has added it recently

Reply:
Then It requires to rearrange all logger severities in  namespace onevpl ( warning ->error/critical). I would implement it in the next PR including other vpl components - not preproc only

Reply:
https://github.com/opencv/opencv/blob/4.x/modules/gapi/src/backends/ie/giebackend.cpp#L728